src/borg/testsuite/hashindex.py

```diff
@@ -181,9 +181,16 @@ def test_size_on_disk_accurate(self):
         idx = ChunkIndex()
         for i in range(1234):
             idx[H(i)] = i, i**2, i**3
-        with tempfile.NamedTemporaryFile() as file:
-            idx.write(file.name)
-            size = os.path.getsize(file.name)
+
+        file, name = tempfile.mkstemp()
+        try:
+            # Close the file, on windows a file cannot be opened multiple times.
+            os.close(file)
+            idx.write(name)
+            size = os.path.getsize(name)
+        finally:
+            os.unlink(name)
```

Comment:
Looks like just giving the `file` (line 184) to `.write` should work and be the preferred solution here.

```
    def write(self, path):
        if isinstance(path, (str, bytes)):
            with open(path, 'wb') as fd:
                hashindex_write(self.index, fd, self.legacy)
        else:
            hashindex_write(self.index, path, self.legacy)
```